agreement_dynamic/models/agreement_section.py

```diff
@@ -44,16 +51,29 @@ def onchange_copyvalue(self):
             self.copyvalue = "${{object.{}.{} or {}}}".format(
                 self.field_id.name,
                 self.sub_model_object_field_id.name,
-                self.default_value or "''",
+                self._get_proper_default_value(),
             )
 
     # compute the dynamic content for jinja expression
     def _compute_dynamic_content(self):
         MailTemplates = self.env["mail.template"]
         for this in self:
             content = MailTemplates._render_template(
-                this.content, "agreement.section", [this.id]
-            )[this.id]
-            this.dynamic_content = content
+                this.content, this.resource_ref_model_id.model, this.res_id,
+            )
+            this.dynamic_content = this._edit_expression(content)
 
+    def _get_proper_default_value(self):
+        self.ensure_one()
+        if not self.default_value:
+            return "''"
```

Comment:
We should still do something with types here - if the type of the field is Integer or Float or Decimal, then it should not include the '', and if it is Char or Text or HTML, then it should.

Have been looking if Odoo has something for that but it doesnt seem so, perhaps:

[repr(field.convert_to_record(None, None))](https://github.com/OCA/OCB/blob/13.0/odoo/fields.py#L1205) can do it.

Reply:
Perhaps I did something stupid, but I could not really figure out how to use field.convert_to_record() in order to check this out. So as you see, I check on field.ttype if it is numerical (int or float actually) and return default value without ''. In all other cases default value gets ''

Reply:
This should be fine, actually the convert_to_read thing was more of a long shot, I found nothing useful but felt I needed to post something here to make up for the time lost looking :-)